### PR TITLE
snapctl: add config in client to disable auth and use it in snapctl

### PR DESCRIPTION
**Pull request by mvo5**
LP: #1660941

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -63,18 +63,28 @@ type Config struct {
 	// BaseURL contains the base URL where snappy daemon is expected to be.
 	// It can be empty for a default behavior of talking over a unix socket.
 	BaseURL string
+
+	// DisableAuth controls if the client should try to read the auth.json
```
Comment (pedronis) [inline client/client.go]: controls whether the client should send an Authorization header from reading the auth.json data.

**Comment by jdstrand** [comment]
I'm not commenting on the code changes, but I was somewhat surprised that the confined config hook tried to use /root/.snap/auth.json instead of /root/snap/$SNAP_NAME/$SNAP_REVISION/.snap/auth.json. Is the config hook not going through 'snap run' for its environment setup? Should it?

**Comment by pedronis** [comment]
@jdstrand the code path (we are supressing here) is for  the "snap"  command itself,  it's not for snaps, it's using $HOME/.snap directly

**Comment by pedronis** [comment]
@jdstrand sorry being imprecise, it's not using $HOME even, it's using the home information from "os/user".Lookup

**Pull request merged**